ccl/backupccl: TestDataDriven failed

ccl/backupccl.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7555758?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7555758?buildTab=artifacts#/) on master @ [ea7c52e5a930be203318db6e338edd5dde7537bd](https://github.com/cockroachdb/cockroach/commits/ea7c52e5a930be203318db6e338edd5dde7537bd):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 324806272 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        ----
        pq: cannot run RESTORE with schema_only until cluster has fully upgraded to 22.2
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2991/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-mixed-version:24:
        upgrade-server [1 args]
        <no input to command>
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2991/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-mixed-version:27:
        exec-sql [0 args]
        RESTORE DATABASE d FROM LATEST IN 'nodelocal://0/full_database_backup/' with schema_only, new_db_name='d2';
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2991/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-mixed-version:32:
        query-sql [0 args]
        SELECT * FROM d2.foo;
        ----
    testutils.go:199: no Invalid Descriptors
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
=== RUN   TestDataDriven/restore-schema-only-multiregion
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2991/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:6:
        new-server [4 args]
        <no input to command>
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2991/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:9:
        exec-sql [0 args]
        CREATE DATABASE d PRIMARY REGION "us-east-1" REGIONS "us-west-1", "eu-central-1";
        CREATE TABLE d.t (x INT);
        INSERT INTO d.t VALUES (1), (2), (3);
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2991/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:15:
        query-sql [0 args]
        SELECT region FROM [SHOW REGIONS FROM DATABASE d] ORDER BY 1;
        ----
        eu-central-1
        us-east-1
        us-west-1
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2991/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:22:
        exec-sql [0 args]
        BACKUP DATABASE d INTO 'nodelocal://1/database_backup/';
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2991/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:26:
        exec-sql [0 args]
        BACKUP INTO 'nodelocal://1/full_cluster_backup/';
        ----
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21534

ccl/backupccl.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7574444?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7574444?buildTab=artifacts#/) on master @ [a035d2e6e7ea57b30115ecc8ee1a5d553a9e3412](https://github.com/cockroachdb/cockroach/commits/a035d2e6e7ea57b30115ecc8ee1a5d553a9e3412):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 309304657 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
=== RUN   TestDataDriven/lock-concurrent-backups
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/lock-concurrent-backups:1:
        new-server [1 args]
        <no input to command>
        ----
=== CONT  TestDataDriven/lock-concurrent-backups
    testutils.go:199: no Invalid Descriptors
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
=== RUN   TestDataDriven/restore-schema-only-multiregion
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:6:
        new-server [4 args]
        <no input to command>
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:9:
        exec-sql [0 args]
        CREATE DATABASE d PRIMARY REGION "us-east-1" REGIONS "us-west-1", "eu-central-1";
        CREATE TABLE d.t (x INT);
        INSERT INTO d.t VALUES (1), (2), (3);
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:15:
        query-sql [0 args]
        SELECT region FROM [SHOW REGIONS FROM DATABASE d] ORDER BY 1;
        ----
        eu-central-1
        us-east-1
        us-west-1
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:22:
        exec-sql [0 args]
        BACKUP DATABASE d INTO 'nodelocal://1/database_backup/';
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:26:
        exec-sql [0 args]
        BACKUP INTO 'nodelocal://1/full_cluster_backup/';
        ----
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7593640?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7593640?buildTab=artifacts#/) on master @ [8357abb668a5adaff781343b394b162fb1b66c6e](https://github.com/cockroachdb/cockroach/commits/8357abb668a5adaff781343b394b162fb1b66c6e):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 308629372 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:7:
        exec-sql [0 args]
        CREATE DATABASE restore;
        CREATE SCHEMA restore.myschema;
        CREATE TABLE foobar (pk int primary key);
        CREATE TABLE restore.myschema.table1 (pk int primary key);
        INSERT INTO restore.myschema.table1 VALUES (1);
        CREATE TYPE data.myenum AS ENUM ('hello');
        ----
    datadriven.go:243: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:16:
        exec-sql [0 args]
        BACKUP INTO 'nodelocal://1/cluster_backup';
        ----
    datadriven.go:243: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:20:
        new-server [4 args]
        <no input to command>
        ----
    datadriven.go:243: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:23:
        exec-sql [0 args]
        SET CLUSTER SETTING storage.mvcc.range_tombstones.enabled = true;
        ----
    datadriven.go:243: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:27:
        exec-sql [0 args]
        SELECT crdb_internal.set_vmodule('lease=3');
        ----
    datadriven.go:243: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:35:
        exec-sql [0 args]
        SET CLUSTER SETTING sql.catalog.descriptor_lease_duration = '1s';
        ----
    datadriven.go:243: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:39:
        exec-sql [0 args]
        SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.after_publishing_descriptors';
        ----
    datadriven.go:243: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:43:
        restore [2 args]
        RESTORE FROM LATEST IN 'nodelocal://0/cluster_backup';
        ----
        job paused at pausepoint
    datadriven.go:243: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-on-fail-or-cancel-retry:48:
        exec-sql [0 args]
        SET CLUSTER SETTING jobs.debug.pausepoints = '';
        ----
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7606036?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7606036?buildTab=artifacts#/) on master @ [cfb5ae9a96e1770daa4aef1615a46e212b561a84](https://github.com/cockroachdb/cockroach/commits/cfb5ae9a96e1770daa4aef1615a46e212b561a84):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 305742214 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:34:
        exec-sql [0 args]
        DROP DATABASE defaultdb;
        CREATE DATABASE defaultdb;
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:39:
        query-sql [0 args]
        SELECT count(*) FROM system.namespace WHERE name = 'defaultdb' AND id > 100
        ----
        1
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:44:
        exec-sql [0 args]
        BACKUP INTO 'nodelocal://1/full_cluster_backup/';
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:48:
        exec-sql [0 args]
        BACKUP Database d INTO 'nodelocal://1/full_database_backup/';
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:54:
        new-server [3 args]
        <no input to command>
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:60:
        exec-sql [0 args]
        RESTORE FROM LATEST IN 'nodelocal://0/full_cluster_backup/' with schema_only, new_db_name='d2';
        ----
        pq: new_db_name can only be used for RESTORE DATABASE with a single target database
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:66:
        exec-sql [1 args]
        CREATE USER testuser
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:71:
        exec-sql [1 args]
        RESTORE FROM LATEST IN 'nodelocal://0/full_cluster_backup/' with schema_only
        ----
        pq: only users with the admin role or the RESTORE system privilege are allowed to perform a cluster restore
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2924/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only:77:
        exec-sql [0 args]
        RESTORE FROM LATEST IN 'nodelocal://0/full_database_backup/' with schema_only;
        ----
        pq: full cluster RESTORE can only be used on full cluster BACKUP files
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7617249?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7617249?buildTab=artifacts#/) on master @ [f0554bc215e31ff53e644936ba645bd895ec0d15](https://github.com/cockroachdb/cockroach/commits/f0554bc215e31ff53e644936ba645bd895ec0d15):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 296946194 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
        exec-sql [2 args]
        SHOW BACKUP 'http://127.0.0.1:39189/testing'
        ----
        pq: only users with the admin role or the EXTERNALIOIMPLICITACCESS system privilege are allowed to access the specified http URI
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/backup-permissions-deprecated:150:
        exec-sql [1 args]
        BACKUP DATABASE d INTO 'nodelocal://0/test3'
        ----
        pq: only users with the admin role or the EXTERNALIOIMPLICITACCESS system privilege are allowed to access the specified nodelocal URI
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/backup-permissions-deprecated:156:
        new-server [3 args]
        <no input to command>
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/backup-permissions-deprecated:159:
        exec-sql [1 args]
        CREATE USER testuser
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/backup-permissions-deprecated:163:
        exec-sql [2 args]
        SHOW BACKUP 'http://127.0.0.1:39189/testing'
        ----
        pq: make storage: external http access disabled
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:199: no Invalid Descriptors
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
    testutils.go:181: Warning: Could not check for invalid descriptors: sql: database is closed
=== RUN   TestDataDriven/restore-permissions
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-permissions:1:
        new-server [1 args]
        <no input to command>
        ----
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestDataDriven [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7636867?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7636867?buildTab=artifacts#/) on master @ [dca415eddac0d659ae6d76b4e3dfdf4076adbd34](https://github.com/cockroachdb/cockroach/commits/dca415eddac0d659ae6d76b4e3dfdf4076adbd34):

Fatal error:

```
panic: test timed out after 59m55s

```
Stack: 

```
goroutine 330019366 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:88:
        exec-sql [0 args]
        BACKUP INTO 'nodelocal://1/no_region_cluster_backup/';
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:92:
        exec-sql [0 args]
        DROP DATABASE no_region_db;
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:96:
        exec-sql [0 args]
        DROP DATABASE no_region_db_2;
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:100:
        exec-sql [1 args]
        SET CLUSTER SETTING sql.defaults.primary_region = 'non-existent-region';
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:104:
        exec-sql [0 args]
        RESTORE DATABASE no_region_db FROM LATEST IN 'nodelocal://1/no_region_database_backup/' with schema_only;
        ----
        pq: region "non-existent-region" does not exist
        HINT: valid regions: eu-central-1, eu-north-1
        --
        set the default PRIMARY REGION to a region that exists (see SHOW REGIONS FROM CLUSTER) then using SET CLUSTER SETTING sql.defaults.primary_region = 'region'
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:112:
        exec-sql [1 args]
        SET CLUSTER SETTING sql.defaults.primary_region = 'eu-central-1';
        ----
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:116:
        exec-sql [0 args]
        RESTORE DATABASE no_region_db FROM LATEST IN 'nodelocal://1/no_region_database_backup/' with schema_only;
        ----
        NOTICE: setting the PRIMARY REGION as eu-central-1 on database no_region_db
        HINT: to change the default primary region, use SET CLUSTER SETTING sql.defaults.primary_region = 'region' or use RESET CLUSTER SETTING sql.defaults.primary_region to disable this behavior
    datadriven_test.go:398: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3840/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/restore-schema-only-multiregion:122:
        query-sql [0 args]
        SHOW DATABASES;
        ----
        defaultdb root <nil> <nil> {} <nil>
        no_region_db root eu-central-1  {eu-central-1} zone
        postgres root <nil> <nil> {} <nil>
        system node <nil> <nil> {} <nil>
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/BUILD.bazel
+++ b/pkg/ccl/backupccl/BUILD.bazel
@@ -180,7 +180,7 @@ go_test(
         "system_schema_test.go",
         "utils_test.go",
     ],
-    args = ["-test.timeout=3595s"],
+    args = ["-test.timeout=7195s"],
     data = glob(["testdata/**"]) + ["//c-deps:libgeos"],
     embed = [":backupccl"],
     shard_count = 16,

--- a/pkg/cmd/generate-bazel-extra/main.go
+++ b/pkg/cmd/generate-bazel-extra/main.go
@@ -284,14 +284,15 @@ unused_checker(srcs = GET_X_DATA_TARGETS)`)
 	}
 }
 
-// excludeSqliteLogicTests removes the sqlite logic tests targets from the
-// given list of targets and returns the updated list.
-func excludeSqliteLogicTests(targets []string) []string {
+// excludeReallyEnormousTargets removes the really enormous test targets
+// from the given list of targets and returns the updated list.
+func excludeReallyEnormousTargets(targets []string) []string {
 	for i := 0; i < len(targets); i++ {
 		var excluded bool
 		for _, toExclude := range []string{
 			"//pkg/ccl/sqlitelogictestccl",
 			"//pkg/sql/sqlitelogictest",
+			"//pkg/ccl/backupccl",
 		} {
 			if strings.HasPrefix(targets[i], toExclude) {
 				excluded = true
@@ -315,9 +316,9 @@ func generateTestsTimeouts() {
 	}
 	for size, timeout := range testSizeToDefaultTimeout {
 		if size == "enormous" {
-			// Exclude sqlite logic tests since they have a custom timeout that
+			// Exclude really enormous targets since they have a custom timeout that
 			// exceeds the default 1h.
-			targets[size] = excludeSqliteLogicTests(targets[size])
+			targets[size] = excludeReallyEnormousTargets(targets[size])
 		}
 		// Let the `go test` process timeout 5 seconds before bazel attempts to kill it.
 		// Note that if this causes issues such as not having enough time to run normally

--- a/pkg/cmd/generate-bazel-extra/main_test.go
+++ b/pkg/cmd/generate-bazel-extra/main_test.go
@@ -47,7 +47,7 @@ func TestParseQueryXML(t *testing.T) {
 	}
 }
 
-func TestExcludeSqliteLogicTests(t *testing.T) {
+func TestExcludeReallyEnormousTests(t *testing.T) {
 	for _, tc := range []struct {
 		in  []string
 		out []string
@@ -55,6 +55,7 @@ func TestExcludeSqliteLogicTests(t *testing.T) {
 		{
 			in: []string{
 				"//pkg/jobs:jobs_test",
+				"//pkg/ccl/backupccl:backupccl_test",
 				"//pkg/sql/sem/eval:eval_test",
 				"//pkg/sql/sqlitelogictest:sqlitelogictest_test",
 			},
@@ -83,6 +84,7 @@ func TestExcludeSqliteLogicTests(t *testing.T) {
 		{
 			in: []string{
 				"//pkg/ccl/sqlitelogictestccl:sqlitelogictestccl_test",
+				"//pkg/ccl/backupccl:backupccl_test",
 				"//pkg/jobs:jobs_test",
 				"//pkg/sql/colexec:colexec_test",
 				"//pkg/sql/sem/eval:eval_test",
@@ -95,6 +97,6 @@ func TestExcludeSqliteLogicTests(t *testing.T) {
 			},
 		},
 	} {
-		require.Equal(t, tc.out, excludeSqliteLogicTests(tc.in))
+		require.Equal(t, tc.out, excludeReallyEnormousTargets(tc.in))
 	}
 }

